modules/ui/fields/wikipedia.js

```diff
@@ -306,6 +307,15 @@ export function uiFieldWikipedia(field, context) {
     }
   }
 
+  function updateEncodedWikiUrl(tagLang, tagArticleTitle, anchor) {
+    const underscoredUrlEncodedTitle = encodeURIComponent(tagArticleTitle.replace(/ /g, '_'));
+    const urlAnchored = toUrlAnchor(anchor);
+    _wikiURL = `${scheme}${tagLang}.${domain}/wiki/${underscoredUrlEncodedTitle}${urlAnchored}`;
+  }
+
+  function toUrlAnchor(anchor) {
+    return anchor ? '#' + anchor : '';
+  }
```

Comment:
This ternary is simple enough to inline. While we’re at it, the anchor should also be encoded. You can apply the same encoding to both the page title and the named anchor, replacing spaces with underscores and calling `encodeURIComponent()`. (There are some [nuanced edge cases](https://www.mediawiki.org/wiki/Manual:PAGENAMEE_encoding) that you probably don’t need to spend time worrying about.)

Reply:
sure thing, will refactor. 

reading through the nuanced edge cases, xin cảm ơn.

Reply:
I came across the fact URI encoding the whole path (that is including the hash/ref of the anchor) causes an unnecessary redirect.

I have refactored the function to keep inline use, and dealt with encoding each part, the title and anchor fragment separately.

Note: I'm unable to manually test via the UI assignment of an anchored wiki URI.

Reply:
I was finally able to manually test with anchored URL, figured out how.